src/Preconditions/ArrayPreconditions.php

```diff
@@ -0,0 +1,68 @@
+<?php namespace Guardsman\Preconditions;
+
+use Guardsman\Exceptions\KeyExists;
+use Guardsman\Exceptions\KeyNotFound;
+use Guardsman\Exceptions\ValueExists;
+use Guardsman\Exceptions\ValueNotFound;
+
+trait ArrayPreconditions
+{
+    /** @return mixed */
+    abstract public function getSubject();
+
+    /**
+     * @throws ValueNotFound if the subject is not a value within the array.
+     *
+     * @return $this
+     */
+    public function isValue(array $array)
```

Comment:
Yeah, I considered something like `containsOnlyClass(Foo::class)` but I'm not sure it would get used. I prefer a `FooCollection` instead with a method of `add` that uses type hinting instead.

My use case for `isValueOf` though is for simple type/status value objects.

```php
class SwitchStatus()
{
    const HIGH = 'high';
    const LOW = 'low';
    const MEDIUM = 'medium';

    private $statuses = [
        self::HIGH,
        self::LOW,
        self::MEDIUM,
    ];

    private $status;

    public function __construct($status) {
        \Guardsman\check($status)->isValueOf($this->statuses);

        $this->status = $status;
    }

    …
```